editor/gui/editor_zoom_widget.cpp

```diff
@@ -141,22 +141,15 @@ void EditorZoomWidget::set_zoom_by_increments(int p_increment_count, bool p_inte
 			}
 		}
 	} else {
-		// Base increment factor defined as the twelveth root of two.
-		// This allows for a smooth geometric evolution of the zoom, with the advantage of
-		// visiting all integer power of two scale factors.
-		// Note: this is analogous to the 'semitone' interval in the music world
-		// In order to avoid numerical imprecisions, we compute and edit a zoom index
-		// with the following relation: zoom = 2 ^ (index / 12)
-
 		if (zoom < CMP_EPSILON || p_increment_count == 0) {
 			return;
 		}
 
-		// zoom = 2**(index/12) => log2(zoom) = index/12
-		float closest_zoom_index = Math::round(Math::log(zoom_noscale) * 12.f / Math::log(2.f));
-
-		float new_zoom_index = closest_zoom_index + p_increment_count;
-		float new_zoom = Math::pow(2.f, new_zoom_index / 12.f);
+		// Zoom is calculated as pow(zoom_factor, zoom_step).
+		// This ensures the zoom will always equal 100% when zoom_step is 0.
+		float zoom_factor = EDITOR_GET("editors/2d/zoom_speed_factor");
```

Comment:
This isn't a super hot path, but it still seems wasteful to re-query this editor setting every time the user zooms.

I would suggest caching it in a member variable, and re-assigning it in `_notification` when `EditorSettings::NOTIFICATION_EDITOR_SETTINGS_CHANGED` is received.

Reply:
Has the impact of fetching project/editor settings ever been measured, to really gauge how severe this is? I was wondering if there's any sort of caching system for settings that are continuously fetched, or if it has to always be manual optimization work.

Reply:
I did it this way because I saw that the integer zoom setting was handled similarly in `CanvasItemEditor::_zoom_callback`. Now I also noticed that `CanvasItemEditor::_gui_input_viewport` (which gets called every time you move/click the mouse) also uses `EDITOR_GET`, plus many other functions. So either it doesn't seem to be a performance issue, or it's something that should be focused on in a separate PR perhaps?

I might still cache the setting however, using `CanvasItemEditor::_update_editor_settings` just like `ViewPanner` does.